cost-analyzer/templates/_helpers.tpl

```diff
@@ -217,3 +217,33 @@ The implied use case is {{ template "cost-analyzer.filterEnabled" .Values }}
 {{- template "cost-analyzer.filter" (dict "v" . "r" $result) }}
 {{- $result | toJson | b64enc }}
 {{- end -}}
+
+{{/*
+This template runs the full check for leader/follower requirements in order to determine
+whether it should be configured. This template will return true if it's enabled and all 
+requirements are met. 
+*/}}
+{{- define "cost-analyzer.leaderFollowerEnabled" }}
+    {{- if .Values.kubecostDeployment }}
+        {{- if .Values.kubecostDeployment.leaderFollower }}
+            {{- if .Values.kubecostDeployment.leaderFollower.enabled }}
+                {{- $replicas := .Values.kubecostDeployment.replicas | default 1 }}
+                {{- if not .Values.kubecostModel.etlFileStoreEnabled }}
+                    {{- "" }}
+                {{- else if (eq (quote .Values.kubecostModel.etlBucketConfigSecret) "") }}
+                    {{- "" }}
+                {{- else if not (gt (int $replicas) 1) }}
+                    {{- ""}}
+                {{- else }}
+                    {{- "true" }}
+                {{- end }}
+            {{- else }}
+                {{- "" }}
+            {{- end }}
+        {{- else }}
+            {{- "" }}
+        {{- end }}
+    {{- else }}
+        {{- "" }}
+    {{- end }}
+{{- end }}
```

Comment:
Helm templating, hooray! 👎

Reply:
I started typing out something I though would look marginally nicer, and it did not look better at all.

Reply:
I would love to think that there is a better way to do this, but each incremental "improvement" I made to the above would break a use-case.